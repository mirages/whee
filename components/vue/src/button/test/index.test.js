import Button from '../'
import { mount } from '@vue/test-utils'

test('the shape prop can set the button shape', async () => {
  const wrapper = mount(Button, {
    props: {
      shape: 'square'
    }
  })
  let html = wrapper.html()
  await wrapper.setProps({ shape: 'round' })
  html += wrapper.html()

  expect(html).toMatchSnapshot()
})

test('the type prop can set the button style types', async () => {
  const wrapper = mount(Button)
  let html = wrapper.html()

  await wrapper.setProps({ type: 'primary' })
  html += wrapper.html()

  await wrapper.setProps({ type: 'danger' })
  html += wrapper.html()

  await wrapper.setProps({ type: 'ghost' })
  html += wrapper.html()

  expect(html).toMatchSnapshot()
})

test('the block prop can set the button display block', async () => {
  const wrapper = mount(Button)
  let html = wrapper.html()

  await wrapper.setProps({ block: true })
  html += wrapper.html()

  expect(html).toMatchSnapshot()
})

test('the disabled prop can set the button disable state', async () => {
  const wrapper = mount(Button)
  let html = wrapper.html()

  await wrapper.setProps({ disabled: true })
  html += wrapper.html()

  expect(html).toMatchSnapshot()
})

test('the loading props can set the button loading state', () => {
  const wrapper = mount(Button, {
    props: {
      loading: true,
      loadingSize: 20,
      loadingType: 'circle'
    }
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('the default slot can set the button text', () => {
  const wrapper = mount(Button, {
    slots: {
      default: () => <p data-test="text">确认无误</p>
    }
  })

  expect(wrapper.find('[data-test=text]').html()).toEqual(
    '<p data-test="text">确认无误</p>'
  )
})
