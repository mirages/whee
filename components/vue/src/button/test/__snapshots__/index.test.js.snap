// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the block prop can set the button display block 1`] = `
<button class="btn btnDefault square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
<button class="btn btnDefault square block">
  <div class="content">
    <div>
    </div>
  </div>
</button>
`;

exports[`the disabled prop can set the button disable state 1`] = `
<button class="btn btnDefault square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
<button class="btn btnDefault square"
        disabled
>
  <div class="content">
    <div>
    </div>
  </div>
</button>
`;

exports[`the loading props can set the button loading state 1`] = `
<button class="btn btnDefault square">
  <div class="content">
    <div class="loading">
      <div data-test="loading-size"
           style="width: 20px; height: 20px;"
      >
        <div class="circle">
          <svg viewbox="0 0 30 30">
            <circle cx="15"
                    cy="15"
                    r="12"
                    data-test="loading-circle"
                    stroke="#4f8efa"
            >
            </circle>
          </svg>
        </div>
      </div>
      <div style="color: rgb(79, 142, 250);">
      </div>
    </div>
    <div>
    </div>
  </div>
</button>
`;

exports[`the shape prop can set the button shape 1`] = `
<button class="btn btnDefault square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
<button class="btn btnDefault round">
  <div class="content">
    <div>
    </div>
  </div>
</button>
`;

exports[`the type prop can set the button style types 1`] = `
<button class="btn btnDefault square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
<button class="btn btnPrimary square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
<button class="btn btnDanger square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
<button class="btn btnGhost square">
  <div class="content">
    <div>
    </div>
  </div>
</button>
`;
